Send delayed sms immediately

gavine99 opened the pull request:
added a 'send now' button to delayed sms send which, when long pressed, aborts the delay and sends the sms immediately.

moved existing 'cancel send' button to far left of screen so it's always in the same ui position for better muscle memory access.

satisfies feature request https://github.com/octoshrimpy/quik/issues/35

![image](https://github.com/user-attachments/assets/de74d832-8e11-4025-8948-2a4d2ca24e0f)

octoshrimpy commented:
I've had thoughts on the UI for this. would it be better to have the send now be a show-on-swipe/longpress of message? that way there aren't a lot of buttons floating around in the convo message. likewise, the cancel button could also be moved there, and just show a lil countdown on bottom right of the message.

gavine99 commented:
agreed about multiple buttons looking ugly.
long press event is already used to select message. short press toggles display/hide status.
swipe is available but wouldn't be evident to users unless there's some sort of instruction on screen.
maybe instead of 'sending...' change status to 'swipe left to cancel, right to send now. sending in X'  with count down  ?

lascapi commented:
Hello, my two cents in one image

![quik-1](https://github.com/user-attachments/assets/b7004f52-b07b-481c-8069-094b4ac7a742)

Like that the button to send is always at the same place.

octoshrimpy commented:
I'm keen on keeping helper text under a to-send message to a minimum. how about "swipe for actions" with the round loader? on swipe, could do something like this mockup, where swipe one way is send now, swipe other is cancel. long press could show both those options as well as a "edit send timedate" option. 
![image](https://github.com/user-attachments/assets/e80d863f-1c1b-4036-a1c2-2a8e81511bed)


this way when scrolling past, the message is mostly unobtrusive without having to dig into menus for all the options. 



a related idea: maybe have a setting as well for "show only scheduled messages coming up within x timeframe"? If I schedule a happy b-day sms for 10 months from now, it makes no sense to have it there every time I open up the convo, but would be nice to have it show up a week before it's sent. Maybe a "3 scheduled" text somewhere? maybe even just an icon next to send? let's spitball something.

lascapi commented:
Hi, i already said what I'm thinking, but I didn't say that at some point you/we should ask a UX designer because the UI become super random! 

Maybe I can ask here : https://opensourcedesign.net/
If the dev team is ready to work for that of course !

gavine99 commented:
the more i think about the hard work involved in all these great ideas, the more i like the original pr ui. heh

just so we're on the same page... this pr is for *delayed* send which means a max of 10 seconds that the send now and cancel buttons are on screen. and if set to 3 or 5 seconds delay it's too short a time to discover to swipe left or right or up or... oh no! my message sent before i remembered which way to slide to cancel it.

![image](https://github.com/user-attachments/assets/c2298050-3e5b-4907-beba-0ae531b6779c)

i think the swiping and setting new schedule stuff is good for *scheduled* messages but could be overkill for delayed.

how about just move the send now button to the rhs for the (max) 10 seconds it's displayed;

![image](https://github.com/user-attachments/assets/6b52b4a2-ffad-4de4-b178-3e00f397e3d3)

octoshrimpy commented:
> i think the swiping and setting new schedule stuff is good for scheduled messages but could be overkill for delayed.


...OH. I misunderstood and thought this was for _scheduled_ messages!! You are absolutely correct.

gavine99 commented:
rebased pr to master to remove conflicts. moved send now button to rhs. made send now button normal click, not long click.
![image](https://github.com/user-attachments/assets/1b6da83c-a3ce-4169-8e9c-5519f95b3879)

gavine99 commented:
button shadows removed. layouts rejiggered to ensure buttons at bottom of long text.

my branch is updated and able to merge to master and hopefully this can sneak into the 4.0.9 beta release.

short text;
![image](https://github.com/user-attachments/assets/9356bbeb-ff93-4175-877a-4592f8c16534)

long text;
![image](https://github.com/user-attachments/assets/adde5b37-3167-425d-a9d2-25a0df0edfd6)
